internal/pkg/cli/app_package_test.go

```diff
@@ -2,3 +2,293 @@
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne("Which application's CloudFormation template would you like to generate?", gomock.Any(), []string{"frontend", "backend"}).Return("frontend", nil)
+			},
+
+			wantedAppName: "frontend",
+			wantedEnvName: "test",
+		},
+		"prompt only for the env name": {
+			inAppName: "frontend",
+
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Times(0)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().ListEnvironments(gomock.Any()).Return([]*archer.Environment{
+					{
+						Name: "test",
+					},
+					{
+						Name: "prod",
+					},
+				}, nil)
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne("Which environment's configuration would you like to use for your stack's parameters?", gomock.Any(), []string{"test", "prod"}).Return("test", nil)
+			},
+
+			wantedAppName: "frontend",
+			wantedEnvName: "test",
+		},
+		"don't prompt": {
+			inAppName: "frontend",
+			inEnvName: "test",
+
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Times(0)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().ListEnvironments(gomock.Any()).Times(0)
+			},
+			expectPrompt: func(m *climocks.Mockprompter) {
+				m.EXPECT().SelectOne(gomock.Any(), gomock.Any(), gomock.Any()).Times(0)
+			},
+
+			wantedAppName: "frontend",
+			wantedEnvName: "test",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockWorkspace := mocks.NewMockWorkspace(ctrl)
+			mockEnvStore := mocks.NewMockEnvironmentStore(ctrl)
+			mockPrompt := climocks.NewMockprompter(ctrl)
+
+			tc.expectWS(mockWorkspace)
+			tc.expectEnvStore(mockEnvStore)
+			tc.expectPrompt(mockPrompt)
+
+			opts := &PackageAppOpts{
+				AppName:  tc.inAppName,
+				EnvName:  tc.inEnvName,
+				ws:       mockWorkspace,
+				envStore: mockEnvStore,
+				prompt:   mockPrompt,
+			}
+
+			// WHEN
+			err := opts.Ask()
+
+			// THEN
+			require.Equal(t, tc.wantedAppName, opts.AppName)
+			require.Equal(t, tc.wantedEnvName, opts.EnvName)
+
+			if tc.wantedErrorS != "" {
+				require.EqualError(t, err, tc.wantedErrorS)
+			} else {
+				require.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestPackageAppOpts_Validate(t *testing.T) {
+	testCases := map[string]struct {
+		inProjectName string
+		inEnvName     string
+		inAppName     string
+
+		expectWS       func(m *mocks.MockWorkspace)
+		expectEnvStore func(m *mocks.MockEnvironmentStore)
+
+		wantedErrorS string
+	}{
+		"invalid workspace": {
+			expectWS: func(m *mocks.MockWorkspace) {
+				m.EXPECT().AppNames().Times(0)
+			},
+			expectEnvStore: func(m *mocks.MockEnvironmentStore) {
+				m.EXPECT().GetEnvironment(gomock.Any(), gomock.Any()).Times(0)
+			},
+
+			wantedErrorS: "could not find a project attached to this workspace, please run `project init` first",
```

Comment:
Nit: For maintainability, should we use the `errNoWorkspace` variable defined in https://github.com/aws/amazon-ecs-cli-v2/pull/260/files#diff-a0a7114926f9397ff0487c11217501deR14?